src/client/common/experiments.ts

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { inject, injectable, named } from 'inversify';
+import { Uri } from 'vscode';
+import { IHttpClient } from '../activation/types';
+import { sendTelemetryEvent } from '../telemetry';
+import { EventName } from '../telemetry/constants';
+import { IApplicationEnvironment, IWorkspaceService } from './application/types';
+import { STANDARD_OUTPUT_CHANNEL } from './constants';
+import { traceDecorators, traceError } from './logger';
+import { ICryptoUtils, IExperimentsManager, IOutputChannel, IPersistentState, IPersistentStateFactory, Resource } from './types';
+import { Experiments } from './utils/localize';
+
+const EXPIRY_DURATION_MS = 30 * 60 * 1000;
+const experimentStorageKey = 'EXPERIMENT_STORAGE_KEY';
+const configUri = 'https://raw.githubusercontent.com/karrtikr/check/master/environments.json';
+
+type AB_Experiments = { name: string; salt: string; min: number; max: number }[];
+@injectable()
+export class ExperimentsManager implements IExperimentsManager {
+    private experimentStorage: IPersistentState<AB_Experiments | undefined>;
+    private activatedWorkspaces = new Map<string, boolean>();
+    private resource: Resource;
+    constructor(
+        @inject(IPersistentStateFactory) private readonly persistentStateFactory: IPersistentStateFactory,
+        @inject(IWorkspaceService) private readonly workspaceService: IWorkspaceService,
+        @inject(IHttpClient) private readonly httpClient: IHttpClient,
+        @inject(ICryptoUtils) private readonly crypto: ICryptoUtils,
+        @inject(IApplicationEnvironment) private readonly appEnvironment: IApplicationEnvironment,
+        @inject(IOutputChannel) @named(STANDARD_OUTPUT_CHANNEL) private readonly output: IOutputChannel
+    ) {
+        this.experimentStorage = this.persistentStateFactory.createGlobalPersistentState<AB_Experiments | undefined>(experimentStorageKey, undefined, EXPIRY_DURATION_MS);
+    }
+
+    public async activate(resource: Uri): Promise<void> {
+        if (this.activatedWorkspaces.has(this.getWorkspacePathKey(resource))) {
+            return;
+        }
+        this.activatedWorkspaces.set(this.getWorkspacePathKey(resource), true);
+        this.resource = resource;
+        this.initializeInBackground().ignoreErrors();
+    }
+
+    @traceDecorators.error('Failed to initialize experiments')
+    public async initializeInBackground() {
+        if (this.isTelemetryDisabled() || this.experimentStorage.value) {
+            return;
+        }
+        const downloadedExperiments = await this.httpClient.getJSONC<AB_Experiments>(configUri);
```

Comment:
Remove `_` from typenames (TypeScript doesn't use `_` in typenames)
```suggestion
        const downloadedExperiments = await this.httpClient.getJSONC<ABExperiments>(configUri);
```